crates/graphql_network_protocol/src/graphql_network_protocol.rs

```diff
@@ -27,18 +27,20 @@ impl NetworkProtocol for GraphQLNetworkProtocol {
 
     type SchemaObjectAssociatedData = GraphQLSchemaObjectAssociatedData;
 
-    fn parse_type_system_document(
+    fn parse_type_system_documents(
         db: &Database,
         schema_source_id: SourceId<SchemaSource>,
-    ) -> Result<MemoRef<Self::TypeSystemDocument>, Box<dyn Error>> {
-        Ok(parse_graphql_schema(db, schema_source_id).to_owned()?)
-    }
-
-    fn parse_type_system_extension_document(
-        db: &Database,
-        schema_extension_source_id: SourceId<SchemaSource>,
-    ) -> Result<MemoRef<Self::TypeSystemExtensionDocument>, Box<dyn Error>> {
-        Ok(parse_schema_extensions_file(db, schema_extension_source_id).to_owned()?)
+        schema_extension_sources: &BTreeMap<RelativePathToSourceFile, SourceId<SchemaSource>>,
+    ) -> Result<
+        MemoRef<(
+            Self::TypeSystemDocument,
+            BTreeMap<RelativePathToSourceFile, Self::TypeSystemExtensionDocument>,
+        )>,
+        Box<dyn Error>,
+    > {
+        parse_graphql_schema(db, schema_source_id, schema_extension_sources)
+            .to_owned()
+            .map_err(|e| Box::new(e) as Box<dyn std::error::Error>)
```

Comment:
just curious, why do you prefer `map_err` over `Ok(parse()?)`

Reply:
Oh, nice, didn't think of that! `Ok(parse()?)` is definitely cleaner